### DISCUSSION TITLE
Ai chatbot learning tips

### DISCUSSION THREAD

**Discussion by markusgous** [community/community#134072] [Programming Help]
### Select Topic Area

General

### Body

Hey there! I am fairly new to git hub. I am currently developing a ai chatbot model. I have load preprocessed datasets for the model I am running with python. I am currently swapping out/comparing different models like Random forest and SGDClassifier to see which accuracy is better. I would appreciate some tips on how to learn the models on the datasets more efficiently as they are learning from the data's x/y train/test sets. The SGDClassifier model is training in batches as incremental learning is only supported with it. I am also vectorizing the text on the training data. The problems I am currently having is the logging the accuracy of the model and the long runtime for the script.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @markusgous ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
